config/source/source.go

```diff
@@ -0,0 +1,40 @@
+// Package source is a base for all config sources.
+package source
+
+import (
+	"fmt"
+	"net/url"
+
+	"go-micro.dev/v5/codecs"
+)
+
+// Data holds a single config file marshaled to map[string]any.
+type Data struct {
```

Comment:
Maybe expand the comment here a little bit to explain exactly how data is used. 
Aka after a config source is read into memory, it will parsed into this data type,
such that every component can process it indivudually, etc...